### DISCUSSION TITLE
grouping sets with rollup/cube

### DISCUSSION THREAD

**Discussion by IcePhoenixby** [linq2db/linq2db#4264] [Q&A]
Hello everyone, can I use cube/rollup inside group by grouping sets? In the t-sql syntax, grouping_set_item is described where you can use cube/rollup like this [ GROUP BY GROUPING SETS ( ROLLUP (Country, Region), CUBE (Country, Region) ); ](https://learn.microsoft.com/en-us/sql/t-sql/queries/select-group-by-transact-sql?view=sql-server-ver16#group-by-grouping-sets--) but  i did't find any examples either in the documentation or in the tests, could you suggest the syntax, or thoughts on how I could expand group by.

**Comment by MaceWindu** [comment]
You can use `Sql.GroupBy.*` helpers. See examples in [tests](https://github.com/linq2db/linq2db/blob/master/Tests/Linq/Linq/GroupByExtensionsTests.cs)

**Reply by IcePhoenixby** [reply]
Could you describe please what it would look like in linq2db.
GROUP BY GROUPING SETS ( (Phone  ,CUBE (Country, Region)) );
 i tried Sql.GroupBy.GroupingSets(x=>{ set1= new { phone,  Sql.GroupBy.Cube(()=>new {Country, Region})  } }) but it does't work, question is how to combine  Sql.GroupBy.Cube/rollup with default group by.

**Reply by MaceWindu** [reply]
Hmm, looks like it is not supported yet. Will convert it to feature request

**Reply by MaceWindu** [reply]
tracked by #4265
